### PR TITLE
[Storage] Support File AD authentication and Kerberos keys

### PR DESCRIPTION
**Pull request by blueww**
## Description

design pass review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/377
(the kerb key in design is single with key name "kerb", but currently there are 2 kerb keys output from server with key name "kerb1", "kerb2", others are same)

The PR not include test, since the feature need domain login, and when try to record test get “Interactive Login is supported only in NET452 and above projects”. Has sent mail for that.

## Checklist

- [x] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Storage/Storage.Management/StorageAccount/SetAzureStorageAccount.cs [hunk 1]
```diff
@@ -274,14 +345,92 @@ public override void ExecuteCmdlet()
...
+
+                        if (enableActiveDirectoryDomainServicesForFile.Value) // Enable AD
+                        {
+                            if (string.IsNullOrEmpty(this.ActiveDirectoryDomainName)
+                                || string.IsNullOrEmpty(this.ActiveDirectoryNetBiosDomainName)
+                                || string.IsNullOrEmpty(this.ActiveDirectoryForestName)
+                                || string.IsNullOrEmpty(this.ActiveDirectoryDomainGuid)
+                                || string.IsNullOrEmpty(this.ActiveDirectoryDomainSid)
+                                || string.IsNullOrEmpty(this.ActiveDirectoryAzureStorageSid)
+                                )
```
Comment (wyunchi-ms) [inline src/Storage/Storage.Management/StorageAccount/SetAzureStorageAccount.cs]: You may want to abstract this into a private function for reusing.

**Comment by blueww** [comment]
@wyunchi-ms 
Thanks for the review and merge!

**Pull request merged**